interactive_shell_handler.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <windows.h>
+#include <psapi.h>
+#include <tchar.h>
+#include "interactive_shell_handler.h"
+#include "utils.h"
+
+// forward declarations
+void perform_command(char *command);
+char** split_command(char *command);
+
+void handle_copy(char* src, char* dst) {
+    int exitCode = 0;
+
+    exitCode = copy_file(src, dst);
+    if (exitCode == 0) {
+        printf("copied source file to destination");
+    } else {
+        printf("failed to copy source file to destination");
+    }
+}
+
+void handle_move(char* src, char* dst) {
+    int exitCode = copy_file(src, dst);
+    if (exitCode == 0) {
+        // delete src file 
+        exitCode = remove(src);
+        if (exitCode == 0) {
+            printf("moved file successfully");
+        } else {
+            printf("failed to moved file");
+        }
+    }
+}
+
+void print_process_name_and_id (DWORD processId) {
+    TCHAR processName[MAX_PATH] = "<unknown>";
+    HANDLE hProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, processId);
+    if (hProcess != NULL) {
+        HMODULE hMod;
+        DWORD cbNeeded;
+
+        if(EnumProcessModulesEx(hProcess, &hMod, sizeof(hMod), &cbNeeded, LIST_MODULES_DEFAULT)) {
+            GetModuleBaseName(hProcess, hMod, processName, sizeof(processName)/sizeof(TCHAR));
+        }
+    }
+
+    printf("%s\t(PID: %u)\n", processName, processId);
```

Comment:
Side note: while not an actual issue, I wouldn't use printf directly but instead implement a LOG macro which would call different underlying functions on DEBUG vs Release builds.
This way the underlying target of the prints is abstracted from the user (and we can log errors more easily).

You can add this in a different branch if you find this interesting enough, don't forget to support multiple arguments to allow format specifiers :)